File: 上學期期末考/[2017AF-B]即刻救援.cpp
#include <bits/stdc++.h>
using namespace std;

#define miyuki_is_my_wife_and_sister ios::sync_with_stdio(false), cin.tie(nullptr)

int main() {
    miyuki_is_my_wife_and_sister;

    int x{};
    bool is_prime = false;

    cin >> x;
    //prime check
    for (int i = 2; i <= sqrt(x); ++i) {
        if (x % i == 0) {
            is_prime = true;
            break;
        }
    }

    if (!is_prime) {
        cout<<"Those whom the gods wish to destroy they first make mad.\n";
        return 0;
    }

    for (int i = x - 1; i >= 1; --i) {
        for (int j = 1; j <= x - 1; ++j) {
            cout << " ";
        }
        cout << i << '\n';
    }

    for (int i = 1; i <= x; ++i) {
        for (int j = 1; j < i; ++j) {
            cout << " ";
        }
        for (int j = (2 * x - 1 - 2 * (i - 1)); j >= 1; --j) {
            cout << "*";
        }
        cout << '\n';
    }

    return 0;
}

File: 上學期期末考/[2017AF-B]即刻救援.py
def main():
    def prime_check(n):
        from math import sqrt, floor
        if n == 2 or n == 3:
            return True
            
        elif n > 4:
            m = n % 6
            
            if m != 1 and m != 5:
                return False
                
            nSqrt = int(floor(sqrt(n)))
            for i in range(5, nSqrt + 1, 6):
                if n % i == 0 or n % (i + 2) == 0:
                    return False
                    
            return True
            
        else:
            return False
    
    x = int(input())
    
    if not prime_check(x):
        print("Those whom the gods wish to destroy they first make mad.")
        return
        
    for i in range(x - 1, 0, -1):
        for j in range(1, x):
            print(" ", end='')
        
        print(i)
    
    for i in range(1, x + 1):
        for j in range(1, i):
            print(" ", end='')
        
        for j in range(2 * x - 1 - 2 * (i - 1), 0, -1):
            print("*", end='')
        
        print()

if __name__ == "__main__":
    # miyuki is my wife and sister

    main()
